ompi/mpi/fortran/mpif-h/init_f.c

```diff
@@ -22,16 +22,7 @@
 
 #include "ompi_config.h"
 
-#if (OPAL_HAVE_WEAK_SYMBOLS || ! OMPI_BUILD_MPI_PROFILING)
-#if OPAL_CC_USE_PRAGMA_IDENT
-#pragma ident OMPI_IDENT_STRING
```

Comment:
Yes, we did this at the request of Sun.  They wanted us to use the `ident` pragma to identify the libraries.

I don't think there was any other hard technical requirement.  

Per https://gcc.gnu.org/onlinedocs/gcc-10.2.0/gnat_rm/Pragma-Ident.html and https://gcc.gnu.org/onlinedocs/gcc-10.2.0/gnat_rm/Pragma-Comment.html, it looks like all this does it add a comment in the generated file.  I don't think it's terribly important since modern packaging systems tend to maintain good control of what files are associated with what packages.

Reply:
ok to remove or should I put it back?

Reply:
I think it was a little more than that, they were exposing this string as a means to detect the existence of some capabilities for their programming environment (debugger and stuff). Anyway, not clear it is still used at this point. If any vendor is still relying on this capability, now is the time to speak up. Otherwise, go ahead and remove it.

Reply:
Nobody has complained - I could separate out this change to its own commit as @bwbarrett  suggested, and leave it on master for a while before bringing it back to v5 to give people longer to chime in.

Reply:
I think I'd still like it in its own commit (even if it is this same PR), with a bit of history on why we removed it in the commit message.  Leaving ourselves notes is a good thing.

Reply:
Done - separated it out. Also rebased on current master.